sdk/Context/Context.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Context;
+
+use OpenTelemetry\Trace as API;
+
+class Context implements API\Context
+{
+    /**
+     * @var ContextKey
+     */
+    private $key;
+
+    /**
+     * @var mixed
+     */
+    private $value;
+
+    /**
+     * @var Context
+     */
+    private $parent;
+
+    /**
+     * Private constructor. All instances should be created via withValue().
+     *
+     * This is heavily influenced by Golang's https://golang.org/pkg/context package
+     */
+    private function __construct(ContextKey $key, $value, ?Context $parent=null)
+    {
+        $this->key = $key;
+        $this->value = $value;
+        $this->parent = $parent;
+    }
+
+    /**
+     * @param ContextKey $key
+     * @param mixed $value
+     * @param Context|null $ctx
+     *
+     * @return Context
+     */
+    public static function withValue(ContextKey $key, $value, ?Context $ctx=null): API\Context
```

Comment:
Maybe we could call this `SetValue` to follow the spec?  That may satisfy:

https://github.com/open-telemetry/opentelemetry-php/issues/105